lib/ruby_lsp/requests/support/dependency_detector.rb

```diff
@@ -44,7 +44,7 @@ def typechecker?
       sig { params(gem_pattern: Regexp).returns(T::Boolean) }
       def direct_dependency?(gem_pattern)
         Bundler.with_original_env { Bundler.default_gemfile } &&
-          Bundler.locked_gems.dependencies.keys.grep(gem_pattern).any?
+          Bundler.locked_gems.specs.map(&:name).grep(gem_pattern).any?
```

Comment:
@vinistock what do you think of this approach? The other way might be to keep `Bundler.locked_gems.dependencies` but separately read from `.gemspec`, then aggregate them.